File: tests/fixtures/ldap_injection/typescript/baseline_constant_ldap.ts
// Baseline: filter is a literal constant; no taint reaches the search call.
import * as ldap from 'ldapjs';

const client = ldap.createClient({ url: 'ldap://example.com' });

export function lookup(_req: any, res: any) {
    const filter = '(objectClass=person)';
    client.search('ou=people,dc=example,dc=com', { filter: filter }, (err: any) => { res.json({ ok: !err }); });
}

File: tests/fixtures/ldap_injection/ruby/baseline_constant_ldap.rb
# Baseline: filter is a literal string, no taint reaches the search call.
require "net/ldap"

class UsersController
  def lookup
    ldap = Net::LDAP.new(host: "ldap.example.com")
    ldap.search(base: "ou=people,dc=example,dc=com", filter: "(objectClass=person)")
  end
end
